### PR TITLE
Calendar tile now shows recurring events

### PR DESCRIPTION
**Pull request by obct537**
Events will have to be re-indexed for this to actually do anything.

### CODE REVIEW

**Comment by obct537** [comment]
@tkimnguyen Any relevant examples you could produce? Never had to write an update step

**Comment by tkimnguyen** [comment]
Is this dead?

**Comment by tkimnguyen** [comment]
@obct537 is the only thing missing here an upgrade step?

**Comment by tkimnguyen** [comment]
Need to figure out why Travis was complaining...

**Comment by vangheem** [comment]
What's the deal with this guy? Close if it's stale and unnecessary anymore!

**Comment by vangheem** [comment]
Closing, reopen if you disagree!

**Comment by tkimnguyen** [comment]
This is still an issue. We were holding off merging so we could test thoroughly (as it adds a new index). Before we reopen this, we need to fix a z-index issue with the event recurrence dialog.

**Comment by tkimnguyen** [comment]
The  upgrade correctly creates the new index and iterates over events that have a recurrence but the new index doesn't end up containing anything... WIP

File: castle/cms/indexers.py [hunk 1]
```diff
@@ -96,6 +96,14 @@ def getContentTypeImage(obj):
         pass
 
 
+@indexer(IEvent)
+def recurrence(obj):
+    try:
+        return obj.recurrence
+    except:
```
Comment (vangheem) [inline castle/cms/indexers.py]: Maybe change to `except AttributeError:`?

File: castle/cms/indexing.zcml [hunk 2]
```diff
@@ -35,6 +35,7 @@
   <adapter name="contentType" factory=".indexers.getContentTypeImage" />
   <adapter name="contentType" factory=".indexers.getContentTypeContent" />
   <adapter name="trashed" factory=".indexers.trashed" />
+  <adapter name="recurrence" factory=".indexers.recurrence" />
```
Comment (vangheem) [inline castle/cms/indexing.zcml]: Do we need an indexer if it's just the name of the attribute?

File: castle/cms/tiles/calendar.py [hunk 3]
```diff
@@ -45,6 +46,19 @@ def json_config(self):
                     'start': brain.start and format_date(brain.start),
                     'end': brain.end and format_date(brain.end)
                 })
+
+                if hasattr(brain, 'recurrence') and brain.recurrence is not None:
```
Comment (vangheem) [inline castle/cms/tiles/calendar.py]: nice!

File: castle/cms/upgrades/upgrade_2_3_3.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+PROFILE_ID = 'profile-castle.cms.upgrades:2_3_3'
+
+
+def upgrade(context, logger=None):
+    setup = getToolByName(context, 'portal_setup')
+    setup.runAllImportStepsFromProfile(PROFILE_ID)
+    # Reindex events that have recurrences
+    brains = api.content.find(portal_type='Event')
+    for brain in brains:
```
Comment (vangheem) [inline castle/cms/upgrades/upgrade_2_3_3.py]: nicely done!

File: castle/cms/profiles/2_3_3/catalog.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0"?>
+<object name="portal_catalog" meta_type="Plone Catalog Tool">
+
+ <index name="recurrence" meta_type="FieldIndex">
```
Comment (vangheem) [inline castle/cms/profiles/2_3_3/catalog.xml]: I think it's only metadata that you are trying to store right?

I think you can remove `<index>` and just leave `<column>`.

Comment (tkimnguyen) [reply castle/cms/profiles/2_3_3/catalog.xml]: yup, seems to work fine. thx!

**Pull request merged**